src/main.ts

```diff
@@ -162,19 +167,21 @@ async function serve(config: webpack.Configuration, args: any) {
 		next();
 	});
 
-	const outputDir = (config.output && config.output.path) || process.cwd();
+	const compiler = args.watch ? await fileWatch(configs, args, true) : await build(configs, args);
+
+	const outputDir = (mainConfig.output && mainConfig.output.path) || process.cwd();
 	let btrOptions = args['build-time-render'];
 	if (btrOptions) {
 		if (args.singleBundle || (args.experimental && !!args.experimental.speed)) {
 			btrOptions = { ...btrOptions, sync: true };
 		}
-		const jsonpName = (config.output && config.output.jsonpFunction) || 'unknown';
+		const jsonpName = (mainConfig.output && mainConfig.output.jsonpFunction) || 'unknown';
 		const onDemandBtr = new OnDemandBtr({
 			buildTimeRenderOptions: btrOptions,
 			scope: libraryName,
 			base,
-			compiler,
-			entries: config.entry ? Object.keys(config.entry) : [],
+			compiler: compiler.compilers[0],
```

Comment:
No need to pass in electron compiler to BTR. Realistically, probably no reason to pass _anything_ to BTR in this case...